packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m

```diff
@@ -190,6 +207,64 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   } else if ([@"DatabaseReference#setPriority" isEqualToString:call.method]) {
     [getReference(call.arguments) setPriority:call.arguments[@"priority"]
                           withCompletionBlock:defaultCompletionBlock];
+  } else if ([@"DatabaseReference#runTransaction" isEqualToString:call.method]) {
+    [getReference(call.arguments) runTransactionBlock:^FIRTransactionResult *_Nonnull(
+                                      FIRMutableData *_Nonnull currentData) {
+      // Handle null value case.
+      if (!currentData.value) {
+        return [FIRTransactionResult successWithValue:currentData];
+      }
+
+      // Create semaphore to allow native side to wait while snapshot
+      // updates occur on the dart side.
+      dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+      // Add semaphore to dictionary so it can be retrieved later.
+      [[self transactionSemaphores] setObject:semaphore forKey:getTransactionKey(call.arguments)];
```

Comment:
I don't think getTransactionKey needs to be a separate function, you can just assign to an `NSString *` and re-use that value in the two places where you need it here.

Reply:
Done!